### PR TITLE
chore: prepare to release 0.2.22

### PR DESCRIPTION
**Pull request by hawkw**
# 0.2.22 (July 20, 2020)

### Fixes
- docs: misc improvements (#2572, #2658, #2663, #2656, #2647, #2630, #2487, #2621,
  #2624, #2600, #2623, #2622, #2577, #2569, #2589, #2575, #2540, #2564, #2567,
  #2520, #2521, #2572, #2493)
- rt: allow calls to `block_on` inside calls to `block_in_place` that are
  themselves inside `block_on` (#2645)
- net: fix non-portable behavior when dropping `TcpStream` `OwnedWriteHalf` (#2597)
- io: improve stack usage by allocating large buffers on directly on the heap
  (#2634)
- io: fix unsound pin projection in `AsyncReadExt::read_buf` and
  `AsyncWriteExt::write_buf` (#2612)
- io: fix unnecessary zeroing for `AsyncRead` implementors (#2525)
- io: Fix `BufReader` not correctly forwarding `poll_write_buf` (#2654)
- io: fix panic in `AsyncReadExt::read_line` (#2541)

### Changes
- coop: returning `Poll::Pending` no longer decrements the task budget (#2549)

### Added
- io: little-endian variants of `AsyncReadExt` and `AsyncWriteExt` methods
  (#1915)
- task: add [`tracing`] instrumentation to spawned tasks (#2655)
- sync: allow unsized types in `Mutex` and `RwLock` (via `default` constructors)
  (#2615)
- net: add `ToSocketAddrs` implementation for `&[SocketAddr]` (#2604)
- fs: add `OpenOptionsExt` for `OpenOptions` (#2515)
- fs: add `DirBuilder` (#2524)

[`tracing`]: https://crates.io/crates/tracing

Signed-off-by: Eliza Weisman <eliza@buoyant.io>

### CODE REVIEW

**Comment by hawkw** [comment]
This is kind of a big one!

File: tokio/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+- io: Fix `BufReader` not correctly forwarding `poll_write_buf` (#2654)
+
+### Changes
+- coop: returning `Poll::Pending` no longer decrements the task budget (#2549)
+
+### Added
+
+- io: little-endian variants of `AsyncReadExt` and `AsyncWriteExt` methods
+  (#1915)
+- io: fix panic in `AsyncReadExt::read_line` (#2541)
```
Comment (Darksonn) [inline tokio/CHANGELOG.md:25]: This seems like it should go in the fixes section.

Comment (hawkw) [reply tokio/CHANGELOG.md:25]: whoops

**Comment by Darksonn** [comment]
This should not be released without #2674.

File: tokio/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,35 @@
+# 0.2.22 (July 20, 2020)
+
+### Fixes
+- docs: misc improvements (#2572, #2658, #2663, #2656, #2647, #2630, #2487, #2621,
+  #2624, #2600, #2623, #2622, #2577, #2569, #2589, #2575, #2540, #2564, #2567,
+  #2520, #2521, #2572, #2493)
```
Comment (Darksonn) [inline tokio/CHANGELOG.md:6]: You have mentioned #2572 twice here.

File: tokio/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,35 @@
+# 0.2.22 (July 20, 2020)
```
Comment (carllerche) [inline tokio/CHANGELOG.md:1]:
```suggestion
# 0.2.22 (July 21, 2020)
```

**Pull request merged**